### ISSUE TITLE
Proper container image?

### ISSUE THREAD

**Issue by sebastian-philipp** [ASKBOT/askbot-devel#801]
I'm coming to the conclusion that deploying Askbot properly is actually quite hard. 
Turns out my existing deployment had svere problems. like missing the cron job, DEBUG=True etc.
I woud not recomment any other kind of deployment. 

I'm starting to thing of making a deployment based on containers the the recommended deployment kind for everyone.

This also makes the Dockerfile the single source of truth for other deployments.

I can think of three usabe deployment setups for Askbot:

**Standard deployment**

* 1 Askbot container (ngnix + uwsgi)
* database container
* 2 volumes (DB + User uploads)

**Extended deployment**

* 3 Askbot containers (ngnix + uwsgi)
* database
* 1 Email cron job
* 1 memcached
* 2 volumes (DB + User uploads)

**Minimal deployment**

* Everthing in one container

Supporting all setups seems feasable with one contaier image.

Is there anything I'm missing?

**Comment by evgenyfadeev** [comment]
perhaps for the minimal container the option might be just the python app with a python server - uwsgi, gunicorn or something similar, mostly for the experts who like minimal per-service containers.

**Comment by sebastian-philipp** [comment]
> perhaps for the minimal container the option might be just the python app with a python server - uwsgi, gunicorn or something similar, mostly for the experts who like minimal per-service containers.

yeah, I'd like to have one container image for all setups. If we're using `manage.py runserver` for just the minimal setup, it gets more complicated.

**Comment by martin-bts** [comment]
With this PR https://github.com/ASKBOT/askbot-devel/pull/819, the image created with `Dockerfile` will work fairly well with sqlite and postgres. For a more versatile image we would have to extend `askbot/container/prestart.py` to accommodate  more DB backends. Right now it only includes a postgres implementation.

**Comment by sebastian-philipp** [comment]
Well, it should work for other backends, but you will then have to set up everything by yourself

**Comment by martin-bts** [comment]
The current image contains all the management commands for setting up the admin user, granting rights and doing migrations. This did/may/will cause trouble if a container does not startup for the very first time or if it is not the first/only Askbot container.

For postgres I implemented a small startup check that will only do what it thinks is required. I am not using Django for this check which makes the code postgres specific. Sqlite, MySQL and Oracle would need their own versions of this startup check.

Further, sqlite and postgres are the only installed Python DB bindings, which is my MySQL and Oracle can't work at the moment.
